libretro-common/include/vfs/vfs.h

```diff
@@ -41,25 +41,39 @@ typedef void* HANDLE;
 #ifdef HAVE_CDROM
 typedef struct
 {
-   char *cue_buf;
+   char* cue_buf;
    size_t cue_len;
-   int64_t byte_pos;
+   size_t cue_pos;
    char drive;
+} vfs_cdrom_t;
+
+typedef struct
+{
+   int64_t byte_pos;
+   unsigned cur_lba;
+   unsigned char cur_track;
    unsigned char cur_min;
    unsigned char cur_sec;
    unsigned char cur_frame;
-   unsigned char cur_track;
-   unsigned cur_lba;
+
+   unsigned sector_size;
+   unsigned char sector_header_size;
+   unsigned char mode;
+
    unsigned last_frame_lba;
    unsigned char last_frame[2352];
-   bool last_frame_valid;
-} vfs_cdrom_t;
+} vfs_cdrom_track_t;
```

Comment:
separated `vfs_cdrom_t` into two structs, one for the cdrom object and one for the track objects. the `vfs_cdrom_track_t` is now allocated instead of being an intrinsic part of the `libretro_vfs_implementation_file` as it contains a sector buffer (2352 bytes) which is unnecessary for all other file types.